tests/tests_fabric/strategies/test_fsdp_integration.py

```diff
@@ -259,6 +266,26 @@ def test_fsdp_load_full_state_dict_into_sharded_model(tmp_path):
     fabric.load(checkpoint_path, state)
     assert state["steps"] == 1
 
+    # Gather all weights and compare
+    with FSDP.summon_full_params(fabric.model, writeback=False, rank0_only=False):
+        params_after = torch.cat([p.cpu().view(-1) for p in fabric.model.parameters()])
+    assert torch.equal(params_before, params_after)
+
+    # Create a raw state-dict checkpoint to test `Fabric.load_raw` too
+    raw_checkpoint_path = checkpoint_path.with_name("model-state-dict")
+    if fabric.global_rank == 0:
+        checkpoint = torch.load(checkpoint_path)
+        torch.save(checkpoint["model"], raw_checkpoint_path)
+    fabric.barrier()
+
+    fabric.run()
+    fabric.load_raw(raw_checkpoint_path, fabric.model)
+
+    # Gather all weights and compare
+    with FSDP.summon_full_params(fabric.model, writeback=False, rank0_only=False):
+        params_after = torch.cat([p.cpu().view(-1) for p in fabric.model.parameters()])
+    assert torch.equal(params_before, params_after)
```

Comment:
I would find it simpler if we mocked `model._forward_module.load_state_dict` so that we can directly assert what was loaded
Instead of creating copies of the params and asserting that they are equal.

Feel free to ignore if you disagree

Reply:
I found that this was the easiest way. Comparing the state dict passed to the mock would be more complicated. Just storing a copy of the flat parameters like I do now doesn't seem so inefficient, and is easy to compare.